dp_check/dp_check.go

```diff
@@ -124,6 +135,88 @@ func getBackendAddrsFromGrpclb(lbAddr string) ([]string, error) {
 	}
 }
 
+func manufacturerReader() (io.Reader, error) {
+	switch runningOS {
+	case "linux":
+		return os.Open(linuxProductNameFile)
+	case "windows":
+		cmd := exec.Command(windowsCheckCommand, windowsCheckCommandArgs)
+		out, err := cmd.Output()
+		if err != nil {
+			return nil, err
+		}
+
+		for _, line := range strings.Split(strings.TrimSuffix(string(out), "\n"), "\n") {
+			if strings.HasPrefix(line, powershellOutputFilter) {
+				re := regexp.MustCompile(windowsManufacturerRegex)
+				name := re.FindString(line)
+				name = strings.TrimLeft(name, ":")
+				return strings.NewReader(name), nil
+			}
+		}
+
+		return nil, errors.New("cannot determine the machine's manufacturer")
+	default:
+		return nil, platformError(runningOS)
+	}
+}
+
+func readManufacturer() ([]byte, error) {
+	reader, err := manufacturerReader()
+	if err != nil {
+		return nil, err
+	}
+	if reader == nil {
+		return nil, errors.New("got nil reader")
+	}
+	manufacturer, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return nil, fmt.Errorf("failed reading %v: %v", linuxProductNameFile, err)
+	}
+	return manufacturer, nil
+}
+
+// isRunningOnGCP checks whether the local system, without doing a network request is
+// running on GCP.
+func isRunningOnGCP() (bool, error) {
+	manufacturer, err := readManufacturer()
+	if os.IsNotExist(err) {
+		return false, err
+	}
+	if err != nil {
+		return false, fmt.Errorf("failure to read manufacturer information: %v", err)
+	}
+	name := string(manufacturer)
+	switch runtime.GOOS {
+	case "linux":
+		name = strings.TrimSpace(name)
+		return name == "Google" || name == "Google Compute Engine", nil
+	case "windows":
+		name = strings.Replace(name, " ", "", -1)
+		name = strings.Replace(name, "\n", "", -1)
+		name = strings.Replace(name, "\r", "", -1)
+		return name == "Google", nil
+	default:
+		return false, platformError(runtime.GOOS)
+	}
+	return false, nil
+}
+
+// hasRouteToLb checks if there is ip route from the current VM to the lb backends
+func hasRouteToLb(iface net.Interface) (bool, error) {
```

Comment:
nit: can we rename this to `hasDirectPathIp6Route`?

`hasRouteToLb` is slightly misleading, because this route is used to access more than just the LBs